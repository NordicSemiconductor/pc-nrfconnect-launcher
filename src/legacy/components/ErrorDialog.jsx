/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import { List } from 'immutable';
import { bool, func, instanceOf } from 'prop-types';

const ErrorDialog = ({ isVisible, messages, onClose }) => (
    <Modal show={isVisible} onHide={onClose}>
        <Modal.Header closeButton>
            <Modal.Title>Error</Modal.Title>
        </Modal.Header>
        <Modal.Body>
            {messages.map(message => (
                <p className="core-error-dialog-message" key={message}>
                    {message}
                </p>
            ))}
        </Modal.Body>
        <Modal.Footer>
            <Button onClick={onClose}>Close</Button>
        </Modal.Footer>
    </Modal>
);

ErrorDialog.propTypes = {
    isVisible: bool.isRequired,
    messages: instanceOf(List).isRequired,
    onClose: func.isRequired,
};

export default ErrorDialog;
