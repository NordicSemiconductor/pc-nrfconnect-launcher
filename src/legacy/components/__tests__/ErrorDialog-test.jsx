/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

/* eslint-disable import/first */

// Do not render react-bootstrap components in tests
jest.mock('react-bootstrap', () => ({
    Modal: 'Modal',
    Button: 'Button',
    ModalHeader: 'ModalHeader',
    ModalFooter: 'ModalFooter',
    ModalBody: 'ModalBody',
    ModalTitle: 'ModalTitle',
}));

import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import ErrorDialog from '../ErrorDialog';

describe('ErrorDialog', () => {
    it('should render hidden dialog', () => {
        expect(
            shallow(
                <ErrorDialog
                    messages={List([])}
                    onClose={() => {}}
                    isVisible={false}
                />
            )
        ).toMatchSnapshot();
    });

    it('should render visible dialog with one message', () => {
        expect(
            shallow(
                <ErrorDialog
                    messages={List(['Oops. An error occurred.'])}
                    onClose={() => {}}
                    isVisible
                />
            )
        ).toMatchSnapshot();
    });

    it('should render visible dialog with two messages', () => {
        expect(
            shallow(
                <ErrorDialog
                    messages={List([
                        'Oops. An error occurred.',
                        'Another error',
                    ])}
                    onClose={() => {}}
                    isVisible
                />
            )
        ).toMatchSnapshot();
    });
});
