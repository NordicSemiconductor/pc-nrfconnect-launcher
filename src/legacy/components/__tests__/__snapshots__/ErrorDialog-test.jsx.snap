// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDialog should render hidden dialog 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Error
    </ModalTitle>
  </ModalHeader>
  <ModalBody />
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`ErrorDialog should render visible dialog with one message 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Error
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <p
      className="core-error-dialog-message"
      key="Oops. An error occurred."
    >
      Oops. An error occurred.
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`ErrorDialog should render visible dialog with two messages 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Error
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <p
      className="core-error-dialog-message"
      key="Oops. An error occurred."
    >
      Oops. An error occurred.
    </p>
    <p
      className="core-error-dialog-message"
      key="Another error"
    >
      Another error
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
